src/llmcompressor/modifiers/autoround/base.py

```diff
@@ -0,0 +1,328 @@
+
+
+def _is_decoding_layer(module, name):
+    return "decoderlayer" in module.__class__.__name__.lower()
+
+
+class _LLModelWrapper(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.layers = torch.nn.ModuleList()
+
+    def forward(self, *args, **kwargs):
+        for layer in self.layers:
+            res = layer(*args, **kwargs)
+        return res
+
+
+class _PretrainModelWrapper(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.model = _LLModelWrapper()
+
+    def forward(self, *args, **kwargs):
+        return self.model(*args, **kwargs)
+
+
+def _wrap_decoding_layer(layer: torch.nn.Module) -> _PretrainModelWrapper:
+    wrapped_model = _PretrainModelWrapper()
+    wrapped_model.model.layers.append(layer)
+    first_param = next(layer.parameters())
+    wrapped_model.dtype = first_param.dtype
+    return wrapped_model
+
+
+class AutoRoundModifier(Modifier, QuantizationMixin):
+    """
+    Implements the AutoRound algorithm from https://aclanthology.org/2024.findings-emnlp.662.pdf.
+    This modifier leverages signed gradient descent (SignSGD) optimizer and
+    block-wise loss to optimize rounding values and weight clipping in a few steps.
+
+    | Sample yaml:
+    | test_stage:
+    |    modifiers:
+    |      AutoRoundModifier:
+    |          iters: 200
+    |          config_groups:
+    |            group_0:
+    |                targets:
+    |                  - "Linear"
+    |                input_activations: null
+    |                output_activations: null
+    |                weights:
+    |                    num_bits: 4
+    |                    type: "int"
+    |                    symmetric: true
+    |                    strategy: group
+    |                    group_size: 128
+
+    Lifecycle:
+        - on_initialize
+            - apply config to model
+        - on_start
+            - add input capture hooks to decoding layers
+        - on_sequential_epoch_end
+            - apply_autoround
+            - post_autoround_cleanup
+        - on_finalize
+            - remove_hooks()
+            - model.apply(freeze_module_quantization)
+
+    :param config_groups: dictionary specifying quantization schemes to apply to target
+        modules. Modules not matching a scheme target will NOT be quantized.
+    :param targets: list of layer names to quantize if a scheme is provided. Defaults
+        to Linear layers
+    :param ignore: optional list of module class names or submodule names to not
+        quantize even if they match a target in config_groups. Defaults to empty list.
+    :param scheme: a single quantization scheme to apply to the model. This is a
+        dictionary that supports all keys from QuantizationScheme except targets, which
+        will be set to the targets parameter set at the modifier level.
+    """
+
+    # AutoRound modifier arguments
+    iters: Optional[int] = 200
+    enable_torch_compile: Optional[bool] = True
```

Comment:
Is there any difference between `enable_torch_compile=False` and `None`? If not, we can simplify this to
```suggestion
    enable_torch_compile: bool = True
```